File: desktop/windows/VoiceBridgeAI/VoiceBridgeAI/Sidecar/AppSupport.cs
namespace VoiceBridgeAI;

public static class AppSupport
{
    public static string DataDirectory
    {
        get
        {
            var appData = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
            return Path.Combine(appData, BundleVariant.AppSupportFolderName);
        }
    }

    public static string ServerLogPath => Path.Combine(DataDirectory, "server.log");

    public static void EnsureLayout()
    {
        Directory.CreateDirectory(DataDirectory);

        var envPath = Path.Combine(DataDirectory, ".env");
        if (File.Exists(envPath))
        {
            return;
        }

        var bundled = BundledSeedText();
        if (bundled is not null)
        {
            File.WriteAllText(envPath, bundled);
            return;
        }

        const string seed = """
            # VoiceBridgeAI — 由 App 自动创建
            AUTO_TEST_ON_START=0
            VOICEBRIDGE_OPTIONAL_LOCAL_MODELS=1
            """;

        File.WriteAllText(envPath, seed);
    }

    private static string? BundledSeedText()
    {
        var path = Path.Combine(AppContext.BaseDirectory, "bundle-seed.env");
        return File.Exists(path) ? File.ReadAllText(path) : null;
    }
}

File: desktop/macos/Sources/VoiceBridgeAI/Sidecar/AppSupport.swift
import Foundation

enum AppSupport {
    static var dataDirectory: URL {
        let base = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first!
        return base.appendingPathComponent(BundleVariant.appSupportFolderName, isDirectory: true)
    }

    static func ensureLayout() throws {
        try FileManager.default.createDirectory(at: dataDirectory, withIntermediateDirectories: true)
        let env = dataDirectory.appendingPathComponent(".env")
        guard !FileManager.default.fileExists(atPath: env.path) else { return }
        if let bundled = bundledSeedText() {
            try bundled.write(to: env, atomically: true, encoding: .utf8)
            return
        }
        let seed = """
        # VoiceBridgeAI — 由 App 自动创建
        AUTO_TEST_ON_START=0
        VOICEBRIDGE_OPTIONAL_LOCAL_MODELS=1
        """
        try seed.write(to: env, atomically: true, encoding: .utf8)
    }

    private static func bundledSeedText() -> String? {
        guard let url = Bundle.main.resourceURL?.appendingPathComponent("bundle-seed.env"),
              FileManager.default.fileExists(atPath: url.path) else { return nil }
        return try? String(contentsOf: url, encoding: .utf8)
    }

    static var serverLogURL: URL {
        dataDirectory.appendingPathComponent("server.log")
    }
}
